### ISSUE TITLE
Issue with generated TryParse in VS 17.5.0 Preview 3.0 - repeated param names

### ISSUE THREAD

**Issue by richardjharding** [SteveDunn/Vogen#335]
### Describe the bug

An existing project upgraded to latest 3.0.10, targeting net7.0 and attempting to build in Vs 17.5.0 preview 3.0 seems to generate incorrect code

SDK version 7.0.102 set by global.json

Any type other than string that needs a TryParse (and its overloads) generates signature like this:

```
public static global::System.Boolean TryParse(string? s s, System.IFormatProvider? provider provider, 
#if NETCOREAPP3_0_OR_GREATER
[global::System.Diagnostics.CodeAnalysis.NotNullWhen(true)]
#endif
 out IacAuthProfileId result) {
```

Note the duplicated "s" and "provider" param names  - this results in VS build reporting errors like below

```
Error	CS0246	The type or namespace name 's' could not be found (are you missing a using directive or an assembly reference?)
Error	CS0246	The type or namespace name 'provider' could not be found (are you missing a using directive or an assembly reference?)
```

This appears to be only a problem in VS - I'm able to build from the command line ok

### Steps to reproduce

create a type with underlying types of int or guide - build from visual studio

### Expected behaviour

try parse methods generated and compile as expected

**Comment by danielcweber** [comment]
I can confirm that this is probably an issue with a preview of VS. Stable VS and CLI works for me as well.

**Comment by SteveDunn** [comment]
Thanks for bug report @richardjharding , and thanks for checking with stable versions @danielcweber .

I'll report the issue to the Roslyn channel on Discord

**Comment by SteveDunn** [comment]
https://discord.com/channels/732297728826277939/735233259763400715/1066639975384371200

**Comment by SteveDunn** [comment]
Should be fixed in version 3.0.11

**Comment by SteveDunn** [comment]
Filed in bug in VS. They can't reproduce it in 17.5.0 Preview 4, so it might be worth trying it there @richardjharding (if you haven't upgraded to Vogen 3.0.11)

https://developercommunity.visualstudio.com/t/1750-Preview-3-breaks-the-Vogen-source/10259243?viewtype=all
